### PR TITLE
std.traits documentation update

### PR DESCRIPTION
**Pull request by burner**
* moved nearly all examples from comments to unittests
** some few had error
* marked existing unittest as examples
** splitted some in the process as they where to long
* ditto for one pair of templates with the same name
* deleted unused code from 2008

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -1768,14 +1752,16 @@ Get the function type from a callable object $(D func).
...
+Example:
+-----------------
 class C
 {
     int value() @property;
 }
 static assert(is( typeof(C.value) == int ));
 static assert(is( FunctionTypeOf!(C.value) == function ));
---------------------
+-----------------
```
Comment (quickfur) [inline std/traits.d]: How come this example isn't in a unittest?

Comment (burner) [inline std/traits.d]: It didn't link as the value property had no body. I added a dummy body and moved it to a unittest.

Comment (quickfur) [inline std/traits.d]: Thanks!

Comment (burner) [inline std/traits.d]: thanks for reviewing. These doc PRs are not the most interesting ones

File: std/traits.d [hunk 2]
```diff
@@ -2507,26 +2334,35 @@ private template hasRawAliasing(T...)
...
+    // struct with a pointer member
+    struct S3 { int a; double * b; }
+    static assert( hasRawAliasing!S3);
+    // struct with an indirect pointer member
+    struct S4 { S3 a; double b; }
+    static assert( hasRawAliasing!S4);
     struct S5 { int a; Object z; int c; }
     static assert( hasRawAliasing!S3);
-    static assert(!hasRawAliasing!S4);
+    static assert( hasRawAliasing!S4);
```
Comment (quickfur) [inline std/traits.d]: Nice catch!

**Comment by andralex** [comment]
thanks

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks for reviewing

File: std/traits.d (hunk 2 above)
Comment (burner) [inline std/traits.d]: easy catch, the compiler told me where to look

**Pull request merged**